Frontend/src/components/Guess.jsx

```diff
@@ -0,0 +1,300 @@
+import React, { useContext, useState } from 'react'
+import Letter from './Letter'
+
+
+
+export default function Guess(props) {
+  const { length, currentGuess } = props
+
+
+
+
+
+  const getBoardLength = () => {
+    if (length == 5) {
+
+
+      return (
+
+        <div className="board flex  items-center">
+          <Letter
+            length={length}
+            currentGuess={currentGuess}
+
+            letterPos={0} />
```

Comment:
Instead of passing in both the `letterPos` and the entire `currentGuess`, just pass in `currentGuess[letterPos]`. That way you're minimizing the _interface_ and the responsibility for what `Letter` needs to be _concerned with_. Both of these improve _separation of concerns_.